### PR TITLE
Helm chart for sentinel-archiver

### PR DESCRIPTION
**Pull request by iand**
This is a helm chart for deploying a [Lily](https://github.com/filecoin-project/lily) node plus [Sentinel Archiver](https://github.com/filecoin-project/sentinel-archiver). See the values file for documentation of configuration.

### CODE REVIEW

**Comment by placer14** [comment]
Just a general thought at first quick look... I think we should make lily's chart [a dependency](https://codefresh.io/docs/docs/new-helm/helm-best-practices/#charts-and-sub-charts) rather than copying it in.

Lily's chart development is still pretty active and would make [changes](https://github.com/filecoin-project/helm-charts/pull/129) painful to propagate.

**Comment by iand** [comment]
> Just a general thought at first quick look... I think we should make lily's chart [a dependency](https://codefresh.io/docs/docs/new-helm/helm-best-practices/#charts-and-sub-charts) rather than copying it in.
> 
> Lily's chart development is still pretty active and would make [changes](https://github.com/filecoin-project/helm-charts/pull/129) painful to propagate.

I don't see how to do that and inject the archiver in a sidecar container. Also, there is a specifiic configuration needed for lily storage config that we don't want to mess with.

File: charts/sentinel-archiver/templates/_helpers.tpl [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    {{- end }}
+    {{- $fingerprint = printf "%st=%s," $fingerprint $taskFragment }}
+  {{- else }}
+  {{- /* Otherwise, fingerprint w first letter of value before and value after */}}
+    {{- $fragment := mustRegexReplaceAll "([a-z0-9])[a-z0-9]*(=?)([0-9]*[a-z]{0,2})" $t "${1}${2}${3}"  }}
+    {{- $fingerprint = printf "%s%s," $fingerprint $fragment }}
+  {{- end }}
+{{- end }}
+{{- $fingerprint }}
+{{- end }}
```
Comment (placer14) [inline charts/sentinel-archiver/templates/_helpers.tpl:121]: Remove unused helper.

Comment (placer14) [inline charts/sentinel-archiver/templates/_helpers.tpl:121]: Remove unused helper.

File: charts/sentinel-archiver/templates/statefulset.yaml [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+      labels:
+        {{- include "sentinel-archiver.allLabels" . | nindent 8 }}
+    spec:
+      # we're going to need root for a little while until
+      # we can package some helper binaries into the image
+      #securityContext:
+        #fsGroup: 532
+        #runAsNonRoot: true
+        #runAsUser: 532
+        #runAsGroup: 532
```
Comment (placer14) [inline charts/sentinel-archiver/templates/statefulset.yaml:25]: Remove comment noise.

File: charts/sentinel-archiver/templates/statefulset.yaml [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+        image: {{ required "(root).image.repo expected" .Values.lily.image.repo }}:{{ default (printf "v%s" .Chart.AppVersion) .Values.lily.image.tag }}
+        imagePullPolicy: {{ .Values.lily.image.pullPolicy }}
+        command: ["/bin/sh", "-c"]
+        args:
+          - |{{- include "sentinel-archiver.chainImportArgs" . | indent 10 }}
+        env:
+        - name: GOLOG_LOG_FMT
+          value: {{ .Values.logFormat | default "json" | quote }}
+        - name: LILY_REPO
+          value: "/var/lib/lily"
```
Comment (placer14) [inline charts/sentinel-archiver/templates/statefulset.yaml:76]: Should we include lily envvars here? There may be some that would apply to import.

**Review by placer14 — APPROVED** [review]
I'm not sure how you want to reconcile this with mainline lily, but maybe we just solve that later. There's already a delta and I believe that there's a way to include the mainline lily chart without repeating ourselves, but we can solve that later.

Looks good otherwise! Sticky 👍

**Comment by placer14** [comment]
Closing stale PR.

**Pull request closed without merging**